### PR TITLE
[test] use bash script to install kubeflow

### PR DESCRIPTION
**Pull request by tejaschumbalkar**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### EIA Checklist
* When creating a PR:
- [ ] I've modified ```src/config/build_config.py``` in my PR branch by setting ```ENABLE_EI_MODE = True```
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*

Use bash script to install kubeflow

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+set -e
+
+install_kfctl(){
+    #install kfctl cli v1.0.2
+    if ! command -v kfctl &> /dev/null
+    then
+        echo "not installed"
+        KFCTL_URL=https://github.com/kubeflow/kfctl/releases/download/v1.0.2/kfctl_v1.0.2-0-ga476281_linux.tar.gz
```
Comment (nskool) [inline test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:9]: Make the kf verison a variable.

File: test/test_utils/eks.py [hunk 2]
```diff
@@ -381,76 +381,27 @@ def eks_setup():
...
     run("aws-iam-authenticator version", echo=True)
-    run("kfctl version", echo=True)
-
 
 def setup_kubeflow(eks_cluster_name,region=os.getenv("AWS_REGION", DEFAULT_REGION)):
-    """Function to setup kubeflow v1.0.2
-        The mxnet operator configuration is not included in the kubeflow v1.0.2 hence installing manually.
-        The mpi operator included in kubeflow v1.0.2 has version v0.1 which has known issues in EKS hence installing the latest
-        version available v0.2
+    """Function to setup kubeflow v1.0.2, MPI and MXNET operators
```
Comment (nskool) [inline test/test_utils/eks.py:393]: avoid version information in docstrings, as it cannot be reliably maintained later.

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+create_dir $1
+
+echo "> Installing kfctl"
+install_kfctl 
+
+echo "> Setting up kubeflow"
+setup_kubeflow $2 
+
+echo "> Installing mxnet operator"
+install_mxnet_operator
```
Comment (saimidu) [inline test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:67]: Is mxnet_operator installed for all frameworks, and not just MXNet?

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh (hunk 1 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh]: done

File: test/test_utils/eks.py (hunk 2 above)
Comment (tejaschumbalkar) [reply test/test_utils/eks.py]: removed

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh (hunk 3 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:69]: This script will install mxnet operator (pod) on eks cluster that would support `MXJob` k8s job.

**Comment by nskool** [comment]
eks tests on the PR are failing - could you update with issue?

**Comment by tejaschumbalkar** [comment]
> eks tests on the PR are failing - could you update with issue?

I changed the kubeflow config file name from kfctl_aws_v1.0.2.yaml → kfctl_aws.yaml 
As the bash script refers to the new file name (in master branch) during execution, it is supposed to fail. 

I have completed the testing locally.

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh (hunk 3 above)
Comment (arjkesh) [reply test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:69]: are there any extra resources allocated with applying mxnet operator?

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+install_mxnet_operator() {
+    #install mxnet operator in EKS cluster
+    git clone https://github.com/kubeflow/mxnet-operator.git
+    kubectl create -k mxnet-operator/manifests/overlays/v1beta1/
+}
+
+create_dir(){
+
+    DIRECTORY="$HOME/$1"
```
Comment (arjkesh) [inline test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:43]: can you give hints as to what the $1 and $2 args are, maybe by assigning them to variables in this file iteself?

File: test/test_utils/eks.py [hunk 5]
```diff
@@ -381,76 +381,27 @@ def eks_setup():
...
-    mpi_operator_dir = os.path.join(home_dir, "mpi-operator")
-    if os.path.isdir(mpi_operator_dir):
-        ctx.run(f"rm -rf {mpi_operator_dir}")
-
-    clone_mxnet_command = f"git clone https://github.com/kubeflow/mpi-operator {mpi_operator_dir}"
-    run(clone_mxnet_command, echo=True)
-    run(f"kubectl create -f {mpi_operator_dir}/deploy/v1alpha2/mpi-operator.yaml", echo=True)
 
+    run(f"chmod +x {local_template_file_path}")
+    run(f"./{local_template_file_path} {eks_cluster_name} {region}",echo=True)
```
Comment (arjkesh) [inline test/test_utils/eks.py:404]: nit: add space between comma and param

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh (hunk 3 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:69]: no explicit resource allocated. https://github.com/kubeflow/mxnet-operator/blob/master/manifests/overlays/v1beta1/kustomization.yaml

File: test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh (hunk 4 above)
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/eks_manifest_templates/kubeflow/install_kubeflow.sh:43]: will add.

**Pull request merged**